group_project_v2/stage/review.py

```diff
@@ -175,9 +207,33 @@ def allowed_nested_blocks(self):
         return blocks
 
     def review_status(self):
-        peer_review_items = self.project_api.get_peer_review_items_for_group(self.workgroup['id'], self.content_id)
+        peer_review_items = self.project_api.get_peer_review_items_for_group(
+            self.workgroup.id, self.activity_content_id
+        )
+
+        return self._check_review_status([user.id for user in self.review_subjects], peer_review_items)
+
+    def get_users_completion(self, target_workgroups, target_users):
+        completed_users, partially_completed_users = set(), set()
 
-        return self._check_review_status(self.review_subjects, peer_review_items, "user")
+        for user in target_users:
+            workgroup = self.project_api.get_user_workgroup_for_course(user.id, self.course_id)
+            review_subjects = set(user.id for user in workgroup.users) - {user.id}
+            review_items = self.project_api.get_peer_review_items_for_group(workgroup.id, self.activity_content_id)
+            grouped_review_items = {
+                self.real_user_id(anonymous_id): grouped_items
+                for anonymous_id, grouped_items in self._group_review_items_by_reviewer(review_items).iteritems()
+            }
```

Comment:
@e-kolpakov Ah, interesting. I just checked into that and see that you are right. I do think you should still look at moving this `grouped_review_items` outside of this loop, maybe to a `grouped_review_items_per_workgroup` cache, so that you only have to create `grouped_review_items` once per unique workgroup instead of once for each user.